mcp/mcp-bundle-api/src/main/java/software/amazon/smithy/mcp/bundle/api/Registry.java

```diff
@@ -11,9 +11,35 @@
 
 public interface Registry {
 
+    final class RegistryEntry {
```

Comment:
This is intentionally not a record. I don't want this to be usable in record destructuring statements.